src/Show/ShowMapper.php

```diff
@@ -68,56 +68,43 @@ public function getAdmin(): AdminInterface
     }
 
     /**
-     * @param FieldDescriptionInterface|string $name
-     *
      * @throws \LogicException
      *
      * @return static
      *
      * @phpstan-param FieldDescriptionOptions $fieldDescriptionOptions
      */
-    public function add($name, ?string $type = null, array $fieldDescriptionOptions = []): self
+    public function add(string $name, ?string $type = null, array $fieldDescriptionOptions = []): self
     {
         if (!$this->shouldApply()) {
             return $this;
         }
 
-        if ($name instanceof FieldDescriptionInterface) {
-            $fieldDescription = $name;
-            $fieldDescription->mergeOptions($fieldDescriptionOptions);
-        } elseif (\is_string($name)) {
-            if (!$this->getAdmin()->hasShowFieldDescription($name)) {
-                $fieldDescription = $this->getAdmin()->createFieldDescription(
-                    $name,
-                    $fieldDescriptionOptions
-                );
-            } else {
-                throw new \LogicException(sprintf(
-                    'Duplicate field name "%s" in show mapper. Names should be unique.',
-                    $name
-                ));
-            }
-        } else {
-            throw new \TypeError(
-                'Unknown field name in show mapper.'
-                    .' Field name should be either of FieldDescriptionInterface interface or string.'
-            );
+        if (isset($fieldDescriptionOptions['role']) && !$this->getAdmin()->isGranted($fieldDescriptionOptions['role'])) {
```

Comment:
In FormMapper this was an early check, I think it's a good idea to have an early return instead of doing all the work and not adding the field.